src/richie/apps/search/index_manager.py

```diff
@@ -112,15 +113,34 @@ def regenerate_indexes(logger):
     # NB: we *must* do this before the update_aliases call so we don't immediately prune
     # version n-1 of all our indexes
     useless_indexes = [
-        index for index, details in existing_indexes.items() if "aliases" not in details
+        index for index, details in existing_indexes.items() if details["aliases"] == {}
```

Comment:
Maybe more robust and nicer ({} creates an empty object for an akward comparison?):
```suggestion
        index for index, details in existing_indexes.items() if not details.get("aliases")
```

Reply:
This will not work as `details["aliases"]` is always defined. When the index is "orphaned" it is an empty object

Reply:
it will work because {} verifies `not details["aliases"]`.
if it is always defined, then:
```suggestion
        index for index, details in existing_indexes.items() if not details["aliases"]
```

Reply:
Thanks, this works great! Done 👍